File: BackEnd_Learn/Java/chat/src/main/java/com/example/chat/Security/Jwt/Auth/AuthDetailsService.java
package com.example.chat.Security.Jwt.Auth;

import com.example.chat.Entity.User.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthDetailsService implements UserDetailsService {

    private final UserRepository userRepository;

    @Override
    public AuthDetails loadUserByUsername(String id) throws UsernameNotFoundException {
        return userRepository.findById(Integer.parseInt(id))
                .map(AuthDetails::new)
                .orElseThrow(RuntimeException::new);
    }
}

File: BackEnd_Learn/kotlin/entity/src/main/kotlin/com/example/entity/security/jwt/auth/AuthDetailsService.kt
package com.example.entity.security.jwt.auth

import com.example.entity.entity.user.UserRepository
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class AuthDetailsService(val userRepository: UserRepository): UserDetailsService {
    override fun loadUserByUsername(id: String): AuthDetails {
        return userRepository.findById(id.toInt())?.let {
            AuthDetails(it) }?: throw UsernameNotFoundException("$id Can Not Found")
    }
}